e2e-polybft/framework/test-server.go

```diff
@@ -189,19 +189,38 @@ func (t *TestServer) Unstake(amount uint64) error {
 }
 
 // RegisterValidator is a wrapper function which registers new validator with given balance and stake
-func (t *TestServer) RegisterValidator(secrets string, balance string, stake string) error {
+func (t *TestServer) RegisterValidator(secrets string, stake string) error {
 	args := []string{
 		"polybft",
 		"register-validator",
-		"--" + polybftsecrets.DataPathFlag, path.Join(t.clusterConfig.TmpDir, secrets),
-		"--registrator-data-dir", path.Join(t.clusterConfig.TmpDir, "test-chain-1"),
+		"--data-dir", path.Join(t.clusterConfig.TmpDir, secrets),
 		"--jsonrpc", t.JSONRPCAddr(),
-		"--balance", balance,
 		"--stake", stake,
 	}
 
 	return runCommand(t.clusterConfig.Binary, args, t.clusterConfig.GetStdout("register-validator"))
 }
+func (t *TestServer) RegisterValidatorNoStake(secrets string) error {
+	args := []string{
+		"polybft",
+		"register-validator",
+		"--data-dir", path.Join(t.clusterConfig.TmpDir, secrets),
+		"--jsonrpc", t.JSONRPCAddr(),
+	}
+
+	return runCommand(t.clusterConfig.Binary, args, t.clusterConfig.GetStdout("register-validator"))
+}
```

Comment:
I'd suggest merging this into RegisterValidator (determine based on `stake` parameter should you provide a stake flag or not).

Reply:
fixed: https://github.com/0xPolygon/polygon-edge/pull/1258/commits/98b6e5890ea7c06fda67307fd63fe1387ddc36cd